fs-dir-nix.go

```diff
@@ -132,7 +132,7 @@ func scandir(dirPath string, filter func(fsDirent) bool, namesOnly bool) ([]fsDi
 				dirent.name = filepath.Join(dirPath, dirent.name)
 			}
 			if dirent.IsDir() {
-				dirent.name += string(os.PathSeparator)
+				dirent.name += string(slashSeparator)
```

Comment:
This won't work as is all the data coming inside should be converted to the 'path' representation and then everywhere we have to use path package functions instead.